warp/src/tesseract/mod.rs

```diff
@@ -478,29 +499,96 @@ impl Tesseract {
     pub fn lock(&self) {
         self.inner.lock();
     }
+}
 
-    /// To save to file using internal file path.
-    /// Note: Because we do not want to interrupt the functions due to it failing to
-    ///       save for whatever reason, this function will return `Result::Ok`
-    ///       regardless of success or error but would eventually log the error.
-    ///
-    /// TODO: Handle error without subjecting function to `Result::Err`
-    pub fn save(&self) -> Result<()> {
-        self.inner.save()
+/// Methods for wasm only
+#[cfg(target_arch = "wasm32")]
+#[cfg_attr(target_arch = "wasm32", wasm_bindgen)]
+impl Tesseract {
+    #[wasm_bindgen(js_name = set)]
+    pub fn set_wasm(&self, key: &str, value: &str) -> std::result::Result<(), JsError> {
+        self.inner.set(key, value).map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = retrieve)]
+    pub fn retrieve_wasm(&self, key: &str) -> std::result::Result<String, JsError> {
+        self.inner.retrieve(key).map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = update_unlock)]
+    pub fn update_unlock_wasm(
+        &self,
+        old_passphrase: &[u8],
+        new_passphrase: &[u8],
+    ) -> std::result::Result<(), JsError> {
+        self.inner
+            .update_unlock(old_passphrase, new_passphrase)
+            .map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = delete)]
+    pub fn delete_wasm(&self, key: &str) -> std::result::Result<(), JsError> {
+        self.inner.delete(key).map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = unlock)]
+    pub fn unlock_wasm(&self, passphrase: &[u8]) -> std::result::Result<(), JsError> {
+        self.inner.unlock(passphrase).map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = save)]
+    pub fn save_wasm(&self) -> std::result::Result<(), JsError> {
+        self.inner.save().map_err(|e| e.into())
+    }
+
+    #[wasm_bindgen(js_name = subscribe)]
+    pub fn subscribe_wasm(&self) -> AsyncIterator {
+        Into::<JsValue>::into(Subscription {
+            inner: self.inner.subscribe(),
+        })
+        .into()
+    }
```

Comment:
I take it that these are renamed to prevent conflicts and to use the `JsError` vs splitting between wasm and non-wasm functions and only using `JsError` under `wasm` target?

Reply:
correct. there are plenty of places that still need to be able to call these from inside the wasm and won't be able to use these js return types.